addons/mail/models/discuss/res_partner.py

```diff
@@ -82,11 +82,11 @@ def get_mention_suggestions_from_channel(self, channel_id, search, limit=8):
             [('user_ids.active', '=', True)],
             [('partner_share', '=', False)]
         ])
-        if channel.group_public_id.id:
+        if channel_group := (channel.parent_channel_id or channel).group_public_id:
```

Comment:
I would not use this operator but define variable in a separate line when the variable is also used outside the scope of the condition.